### ISSUE TITLE
bug with merging `cacheGroups`

### ISSUE THREAD

**Issue by evilebottnawi** [survivejs/webpack-merge#145]
Reproducible test repo: https://github.com/artem-malko/webpack-5-beta-config-bug

Ref: https://github.com/webpack/webpack/issues/11396

Simple reproducible example

```js
console.log(
  merge(
    { 
      optimization: {
        runtimeChunk: {
          name: 'runtime',
        },
    
        splitChunks: {
          minChunks: 2,
              minSize: 30000,
      
              cacheGroups: {
            clientApplication: {
              name: 'clientApplication',
                  test: /applications\/client/,
                  chunks: 'all',
                  enforce: true,
            },
          },
        },
      }
    },
    {
      optimization: {
        runtimeChunk: {
          name: 'runtime',
        },

        splitChunks: {
          minChunks: 2,
          minSize: 30000,

          cacheGroups: {
            clientApplication: {
              name: 'clientApplication',
              test: /applications\/client/,
              chunks: 'all',
              enforce: true,
            },
          },
        },
      }
    }
  ).optimization?.splitChunks
);
```

`test` is `Object`

**Comment by bebraw** [comment]
Thanks for the report. The problem was that there was a check missing against merging a `RegExp` against another `RegExp`. In this case in the configuration definition it was set up so that the `optimization` part was merged to itself `optimization` due to a `...` elsewhere in the configuration.

@artem-malko I've fixed the issue in `5.1.3` but I would recommend against doing `...clientConfig.optimization,` in your production configuration as that will cause the same configuration to appear **twice** although it will be merged away. Can you explain why you did your own merge for that particular configuration? What's the use case?

**Comment by artem-malko** [comment]
@bebraw Hm, nice catch) May be it's because where was no any webpack-merge many years ago) So, we merged our configs manually) 

You're right! I'll remove `...clientConfig.optimization` from my config) optimization part will be used without it too) Thx
